src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -0,0 +1,9260 @@
+ * genProduceReg() on the target register.
+ */
+void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTree* tree)
+{
+    switch (tree->gtOper)
+    {
+        case GT_CNS_INT:
+        {
+            // relocatable values tend to come down as a CNS_INT of native int type
+            // so the line between these two opcodes is kind of blurry
+            GenTreeIntConCommon* con    = tree->AsIntConCommon();
+            ssize_t              cnsVal = con->IconValue();
+
+            // if (con->ImmedValNeedsReloc(compiler))
+            if (con->ImmedValNeedsReloc(compiler) && compiler->opts.compReloc)
+            {
+                // instGen_Set_Reg_To_Imm(EA_HANDLE_CNS_RELOC, targetReg, cnsVal);
+                assert(compiler->opts.compReloc);
+                GetEmitter()->emitIns_R_AI(INS_bl, EA_HANDLE_CNS_RELOC, targetReg, cnsVal);
+                regSet.verifyRegUsed(targetReg);
+            }
+            else
+            {
+                genSetRegToIcon(targetReg, cnsVal, targetType);
+            }
+        }
+        break;
+
+        case GT_CNS_DBL:
+        {
+            emitter* emit       = GetEmitter();
+            emitAttr size       = emitActualTypeSize(tree);
+            double   constValue = tree->AsDblCon()->gtDconVal;
+
+            // Make sure we use "daddiu reg, zero, 0x00"  only for positive zero (0.0)
+            // and not for negative zero (-0.0)
+            if (*(__int64*)&constValue == 0)
+            {
+                // A faster/smaller way to generate 0.0
+                // We will just zero out the entire vector register for both float and double
+                emit->emitIns_R_R(INS_movgr2fr_d, EA_8BYTE, targetReg, REG_R0);
+            }
+            /*else if (emitter::emitIns_valid_imm_for_fmov(constValue))
+            {// LOONGARCH64 doesn't need this.
+                assert(!"unimplemented on LOONGARCH yet");
+            }*/
+            else
+            {
+                // Get a temp integer register to compute long address.
+                // regNumber addrReg = tree->GetSingleTempReg();
+
+                // We must load the FP constant from the constant pool
+                // Emit a data section constant for the float or double constant.
+                CORINFO_FIELD_HANDLE hnd = emit->emitFltOrDblConst(constValue, size);
+
+                // Load the FP constant.
+                assert(targetReg >= REG_F0);
+
+                instruction ins = size == EA_4BYTE ? INS_fld_s : INS_fld_d;
+
+                // Compute the address of the FP constant and load the data.
+                emit->emitIns_R_C(ins, size, targetReg, REG_NA, hnd, 0);
+            }
+        }
+        break;
+
+        default:
+            unreached();
+    }
+}
+
+// Produce code for a GT_INC_SATURATE node.
+void CodeGen::genCodeForIncSaturate(GenTree* tree)
+{
+    assert(!"unimplemented on LOONGARCH yet");
+#if 0
```

Comment:
I would argue against introducing the dead code under `#if 0` here and in other places unless it is strictly necessary.
Can you consider removing this?